src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy

```diff
@@ -61,6 +64,48 @@ class BidValidationSpec extends BaseSpec {
                        }]
     }
 
+    def "PBS should contain response and emit warning logs when bidRequest include  multiple distribution channel"() {
+        given: "Start time and random referer"
+        def startTime = Instant.now()
+        def randomReferer = PBSUtils.randomString
+
+        and: "Request distribution channels"
+        def requestDistributionChannels = bidRequest.getRequestDistributionChannels()
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest, ["Referer" : randomReferer])
+
+        then: "BidResponse contain single seatbid"
+        assert response.seatbid.size() == 1
+
+        then: "Response should contain warning"
+        def warningLogMessages = requestDistributionChannels.collect { "${it.value.toLowerCase()}" }.join(" and ")
+        assert response.ext?.warnings[ErrorType.PREBID]*.code == [999]
+        assert response.ext?.warnings[ErrorType.PREBID]*.message ==
+                ["BidRequest contains $warningLogMessages. Only the first one is applicable, the others are ignored" as String]
+
+        and: "PBS log should contain message"
+        def logs = defaultPbsService.getLogsByTime(startTime)
+        def validatorLogMessage = requestDistributionChannels.collect { "request.${it.value.toLowerCase()}" }.join(" and ")
+        assert getLogsByText(logs, "o.p.server.validation.RequestValidator   : $validatorLogMessage are present. Referer: $randomReferer")
+        assert getLogsByText(logs, "o.prebid.server.auction.ExchangeService  : $warningLogMessages are present. Referer: $randomReferer. Account: ${bidRequest.getAccountId()}")
```

Comment:
Can be `assert getLogsByText(logs, "$validatorLogMessage are present. Referer: $randomReferer")`